codegen/tests/trait_generic_param.rs

```diff
@@ -23,7 +23,7 @@ trait Versioned<const V: u8> {
 }
 
 #[delegate(derive(
-    for<> Named<String>;
+    Named<String>;
     for<const V: u8> Versioned<{ V }>;
 ))]
 enum Users {
```

Comment:
@50U10FCA7 this is the only valid case for using `for<>` syntax: when we need to parametrize the derived trait with type parameters that are absent in the type declaration.

Reply:
@tyranron Here is the same situation as described [above](https://github.com/arcane-rs/delegation/pull/10#discussion_r1924008066). We should not erase any generics, because `Users` doesn't have any.

Reply:
@50U10FCA7 I didn't highlight this case as a problem. Quite opposite: I emphasized it to be the only valid case.